cmd/data-crawler.go

```diff
@@ -488,30 +507,119 @@ func (f *folderScanner) scanQueuedLevels(ctx context.Context, folders []cachedFo
 
 			// Dynamic time delay.
 			t := UTCNow()
-
-			err = objAPI.HealObjects(ctx, bucket, prefix, madmin.HealOpts{Recursive: true, Remove: healDeleteDangling},
-				func(bucket, object, versionID string) error {
-					// Wait for each heal as per crawler frequency.
+			resolver.bucket = bucket
+
+			foundObjs := false
+			dangling := true
```

Comment:
Can't we do the reverse ? set dangling to false here and set to true only in specific cases ? it is easier that way

Reply:
@klauspost ^^

Reply:
@vadmeste How would that work logically? We cannot know from a single returned object/folder if the entire thing is dangling.

Right now dangling will be marked false if any of these are true:

* Any disk returns an error.
* Any Object with read quorum is found.
* If the call returns any error.

This of course all relies on the underlying scans and resolution checks to work as we expect, but I don't see any way around that.